src/core/datastore/sync/sync-manager.js

```diff
@@ -291,7 +291,6 @@ export class SyncManager {
   }
 
   _fetchItemsFromServer(collection, query, options) {
```

Comment:
Wouldn't this result in redundancies? By "this" I mean

```javascript
return this._fetchItemsFromServer(collection, query, options)
      .then(entities => this._replaceOfflineEntities(collection, query, entities))
      .then(replacedEntities => replacedEntities.length);
```

in `SyncManager` (it's not in the diff). This is the old code and it seems like the updating and deleting (the `_replaceOfflineEntities()` part) is now being done inside the deltafetch "module". Same goes for when doing a CacheStore.read(). The old `DeltaFetchRequest` only did a read, the new one does writes too. I guess something else has to change to reflect this too. These redundant deletes and updates would result in worse performance - they are very expensive since we're working in memory.

Reply:
Actually, it's just the updating that's always done. The deleting is done only for when doing a delta set request.

Reply:
Should something be changed related to this?

Reply:
I made a change related to this.

Reply:
Should we be adding this same check to `CacheStore.find()`?